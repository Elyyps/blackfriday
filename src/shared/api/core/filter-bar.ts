import { generateDummyCheckboxArray, Checkbox } from "./checkbox";

export class FilterBar {
  public brands: Checkbox[];
  public categories: string[];
  public sortBy: string[];
  public status: string[];
}

export const generateDummyFilterBar = (): FilterBar => ({
  status: ["Stock", "out", "unlimited"],
  categories: ["A", "B", "C", "D", "E", "F", "G", "H", "I"],
  brands: generateDummyCheckboxArray(),
  sortBy: ["New", "Relevant"]
});
