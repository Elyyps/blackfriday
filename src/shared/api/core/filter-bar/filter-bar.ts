import { Checkbox } from "../checkbox";

export class FilterBar {
  public brandFilterPlaceholder: string;
  public brandFilterTitle: string;
  public brands: Checkbox[];
  public categories: Checkbox[];
  public categoryFilterPlaceholder: string;
  public categoryFilterTitle: string;
  public sortBy: string[];
  public sortByFilterTitle: string;
  public status: Checkbox[];
  public statusFilterTitle: string;
}
