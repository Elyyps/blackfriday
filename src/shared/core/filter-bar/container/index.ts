export * from "./filter-bar.container";
