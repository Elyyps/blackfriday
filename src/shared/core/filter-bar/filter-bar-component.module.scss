.filter-bar {
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.08);
  padding: 16px 0;

  &__filter-container {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
  }
  &__clear-filter {
    margin-left: 20px;
    margin-top: 8px;
    @media (max-width: 820px) {
      margin-left: 10px;
    }
  }
  // &-ul {
  //   list-style: none;
  //   width: 230px;
  // }
  // &-li {
  //   display: flex;
  //   justify-content: space-between;
  //   align-items: center;
  // }
  // &-li:hover {
  //   cursor: pointer;
  // }
  &__filter-list {
    display: flex;
    flex-direction: row;
    width: 65%;
    @media (max-width: 1060px) {
      width: 90%;
    }
    > * {
      padding: 0 4px;
    }
    .filter {
      &-content {
        margin: 0;
        display: none;
        position: relative;
        &-isActive {
          display: block;
        }
        &-label {
          min-height: 44px;
          box-sizing: border-box;
          line-height: 1;
          display: flex;
          justify-content: space-between;
          padding: 12px 16px;
          background-color: $gray-dark;
          color: $global-inverse-color;
          // position: relative;
          // align-items: center;
          &.arrowAction {
            justify-content: center;
          }
          > span {
            margin-left: 8px;
          }
          .button-control,
          .arrow-control {
            .icon-svg {
              transform: rotate(0deg);
              margin: 0;
              path {
                stroke: $global-inverse-color;
              }
              &:hover,
              &:active {
                cursor: pointer;
                outline: none;
              }
            }
          }
          .arrow-control {
            position: absolute;
            left: 0;
            top: 50%;
            width: 100%;
            background-color: transparent;
            border: none;
            line-height: 8;
            padding: 16px;
            transform: translateY(-50%);
            &:hover,
            &:active {
              cursor: pointer;
              outline: none;
            }
          }
          &-icon {
            margin-left: 8px;
            transform: rotate(90deg);
          }
        }
        &-clear {
          text-decoration: underline;
          font-size: 13px;
          font-weight: 600;
          color: $tint-gray;
          &:hover {
            text-decoration: none;
          }
        }
      }
      &-label {
        @media (max-width: 800px) {
          display: none;
        }
        padding-right: 12px;
        font-size: 14px;
        font-weight: 600;
        display: flex;
        align-items: center;
        font-family: $global-font-family-secondary;
        color: $gray-dark;
        > span {
          margin-left: 8px;
        }
        &--mobile {
          background-color: $tint-gray-light;
          justify-content: space-between;
          span {
            transform: rotate(0) translateX(-2px);
            path {
              fill: $gray-dark;
            }
          }
        }
      }
      &-item {
        padding: 2px 4px 0 4px;
        overflow: hidden;
      }
      &-action {
        position: relative;
        display: flex;
        justify-content: space-between;
      }
    }
  }
  &__sort {
    display: flex;
    align-items: center;
    justify-content: flex-end;
  }
  .filter__sort {
    &-item {
      display: flex;
      align-items: center;
      font-size: 15px;
      color: $gray-light;
    }
    &-filter {
      display: flex;
      align-items: center;
      font-size: 15px;
      color: $gray-light;
      padding-left: 15px;
      @media (max-width: 1060px) {
        display: none;
      }
    }
    &-text {
      @media (max-width: 1060px) {
        display: none;
      }
    }
  }
}

.amount-of-shops {
  margin-right: 12px;
  @media (max-width: 1000px) {
    margin-right: 12px;
  }
}
