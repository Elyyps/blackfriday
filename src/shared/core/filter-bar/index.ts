export * from "./filter-bar.component";
export * from "./container";
