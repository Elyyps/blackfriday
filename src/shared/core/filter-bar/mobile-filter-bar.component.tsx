import React, { useState, useEffect } from "react";

import { IMobileFilterSelectedItems } from "../mobile-filter";
import { FilterItem } from "@app/api/core/filter/filter-item";
import { IMobileFilterItem } from "../mobile-filter/mobile-filter-item";
import { MobileFilterContainer } from "../mobile-filter/containers/mobile-filter-container";

export interface IStoresMobileFilterBarComponentProps {
  brandFilterItems: FilterItem[];
  categoryFilterItems: FilterItem[];

  onFiltersChanged: (
    brandFilterItems: FilterItem[],
    categoryFilterItems: FilterItem[],
    statusFilterItem: FilterItem[],
    sortBy: string
  ) => void;
  sortBy: string;
  sortByOptions: string[];
  statusFilterItems: FilterItem[];
}

const StoresMobileFilterBarComponent = (props: IStoresMobileFilterBarComponentProps) => {
  const [mobileFilters, setMobileFilters] = useState<IMobileFilterItem[]>([]);
  const [totalStores, setTotalStores] = useState<number>(0);

  useEffect(() => {
    setMobileFilters(
      getMobileFilterItems(
        props.brandFilterItems,
        props.categoryFilterItems,
        props.sortBy,
        props.sortByOptions,
        props.statusFilterItems
      )
    );
  }, [props.brandFilterItems, props.categoryFilterItems, props.sortBy, props.sortByOptions, props.statusFilterItems]);

  const calculateStoreNumbers = (selectedItems: IMobileFilterSelectedItems[]) => {
    let count = 0;
    selectedItems.map(item => {
      if (item.title === "Merken") {
        props.brandFilterItems.forEach((filter: FilterItem) => {
          if (item.selectedItems.includes(filter.displayName)) {
            if (filter.totalAmount) count += filter.totalAmount;
          }
        });
      }
      if (item.title === "Categorieën") {
        props.categoryFilterItems.forEach((filter: FilterItem) => {
          if (item.selectedItems.includes(filter.displayName)) {
            if (filter.totalAmount) count += filter.totalAmount;
          }
        });
      }
      if (item.title === "Status") {
        props.statusFilterItems.forEach((filter: FilterItem) => {
          if (item.selectedItems.includes(filter.displayName)) {
            if (filter.totalAmount) count += filter.totalAmount;
          }
        });
      }
    });
    setTotalStores(count);
  };
  const handleFinishSearch = (selectedItems: IMobileFilterSelectedItems[]) => {
    const brandFilters = props.brandFilterItems.map((filter: FilterItem) => {
      const result = { ...filter, isSelected: false };

      return result;
    });
    const categoryFilters = props.categoryFilterItems.map((filter: FilterItem) => {
      const result = { ...filter, isSelected: false };

      return result;
    });
    const statusFilters = props.statusFilterItems.map((filter: FilterItem) => {
      const result = { ...filter, isSelected: false };

      return result;
    });

    let sortBy = "";

    selectedItems.map(item => {
      if (item.title === "Merk") {
        brandFilters.forEach((filter: FilterItem) => {
          if (item.selectedItems.includes(filter.displayName)) {
            filter.isSelected = true;
          }
        });
      }
      if (item.title === "Categorieën") {
        categoryFilters.forEach((filter: FilterItem) => {
          if (item.selectedItems.includes(filter.displayName)) {
            filter.isSelected = true;
          }
        });
      }
      if (item.title === "Status") {
        statusFilters.forEach((filter: FilterItem) => {
          if (item.selectedItems.includes(filter.displayName)) {
            filter.isSelected = true;
          }
        });
      }
      if (item.title === "Sorteren") {
        sortBy = item.selectedItems.length > 0 ? item.selectedItems[0] : "Relevatie";
      }
    });

    props.onFiltersChanged(brandFilters, categoryFilters, statusFilters, sortBy);
  };

  const clearAllFilters = () => {
    const clearedBrandFilters = props.brandFilterItems.map((filter: FilterItem) => {
      const result = { ...filter, isSelected: false };

      return result;
    });
    const clearedCategoryFilters = props.categoryFilterItems.map((filter: FilterItem) => {
      const result = { ...filter, isSelected: false };

      return result;
    });
    const clearedStatusFilters = props.statusFilterItems.map((filter: FilterItem) => {
      const result = { ...filter, isSelected: false };

      return result;
    });
    setTotalStores(0);
    props.onFiltersChanged(clearedBrandFilters, clearedCategoryFilters, clearedStatusFilters, "Relevantie");

  };

  return (
    <MobileFilterContainer
      onFilterChange={calculateStoreNumbers}
      onFinish={handleFinishSearch}
      totalStores={totalStores}
      onClear={clearAllFilters}
      filterItems={mobileFilters}
    />
  );
};

const getMobileFilterItems = (
  brandfilterItems: FilterItem[],
  categoryFilterItems: FilterItem[],
  selectedSortBy: string,
  sortByOptions: string[],
  statusFilterItems: FilterItem[]
): IMobileFilterItem[] => [
  {
    hasSearchBar: false,
    title: "Sorteren",
    searchBarPlaceholder: "Sorteren zoeken",
    selectedItems: [selectedSortBy],
    isSingleSelection: true,
    items: sortByOptions
  },
  {
    hasSearchBar: true,
    title: "Status",
    searchBarPlaceholder: "Status zoeken",
    selectedItems: [...getSelectedFilterItems(statusFilterItems)],
    items: [...getFilterItems(statusFilterItems)]
  },
  {
    hasSearchBar: true,
    title: "Categorieën",
    searchBarPlaceholder: "Categorieën zoeken",
    selectedItems: [...getSelectedFilterItems(categoryFilterItems)],
    items: [...getFilterItems(categoryFilterItems)]
  },
  {
    hasSearchBar: true,
    title: "Merken",
    searchBarPlaceholder: "Merk zoeken",
    selectedItems: [...getSelectedFilterItems(brandfilterItems)],
    items: [...getFilterItems(brandfilterItems)]
  }
];

const getFilterItems = (items: FilterItem[]): string[] => items.map(item => item.displayName);

const getSelectedFilterItems = (items: FilterItem[]): string[] => {
  const selectedItems = items.filter(item => item.isSelected);

  return selectedItems.map(item => item.displayName);
};

export { StoresMobileFilterBarComponent };
