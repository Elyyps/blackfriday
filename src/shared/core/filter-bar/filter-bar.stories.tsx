import React from "react";
import { shallow } from "enzyme";
import { FilterBarComponent } from "./filter-bar.component";

describe("[FilterBar]", () => {
  it("should render component without crashing", () => {
    // const renderedComponent = shallow(<FilterBarComponent />);
    // expect(renderedComponent).toMatchSnapshot();
  });
});
