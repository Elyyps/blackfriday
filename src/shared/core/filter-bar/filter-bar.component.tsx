import React from "react";
import styles from "./filter-bar-component.module.scss";
import HandPointing from "@assets/icons/hand-pointing.svg";
import StoreIcon from "@assets/icons/store.svg";
import { IconComponent } from "@app/core";
import { CheckboxDropdown } from "../checkbox-dropdown/checkbox-dropdown.component";
import { FilterItem } from "@app/api/core/filter/filter-item";
import { SearchableCheckboxDropdown } from "../searchable-checkbox-dropdown/searchable-checkbox-dropdown.component";
import { SelectComponent } from "../select";
import { FilterBarContainerProps } from "./container";
import { ViewType } from "@app/stores/settings";
import { StoresMobileFilterBarComponent } from "./mobile-filter-bar.component";
import { injectIntl, InjectedIntlProps } from "react-intl";

export interface IFilterBarProps {
  filtersChanged: () => void;
}

const component = (props: IFilterBarProps & FilterBarContainerProps & InjectedIntlProps) => {
  const onStatusFilterItemsChanged = (items: FilterItem[]) => {
    if (getNumberOfSelectedfilters(items) !== getNumberOfSelectedfilters(props.statusFilterItems)) {
      props.setStatusFilters([...items]);
      props.filtersChanged();
    }
  };

  const onCategoryFilterItemsChanged = (items: FilterItem[]) => {
    if (getNumberOfSelectedfilters(items) !== getNumberOfSelectedfilters(props.categoryFilterItems)) {
      props.setCategoryFilters([...items]);
      props.filtersChanged();
    }
  };

  const onBrandFilterItemsChanged = (items: FilterItem[]) => {
    if (getNumberOfSelectedfilters(items) !== getNumberOfSelectedfilters(props.brandFilterItems)) {
      props.setBrandFilters([...items]);
      props.filtersChanged();
    }
  };

  const setSort = (sortByString: string) => {
    props.setSortBy(sortByString);
  };

  const sortByOptions = ["Relevatie", "Nieuwste", "Populair", "Winkels A - Z", "Winkels Z - A"];

  const getTotalNumberOfFilters = (): number =>
    props.statusFilterItems.filter(item => item.isSelected).length +
    props.categoryFilterItems.filter(item => item.isSelected).length +
    props.brandFilterItems.filter(item => item.isSelected).length;

  const mobileFiltersChanged = (
    brandFilterItems: FilterItem[],
    categoryFilterItems: FilterItem[],
    statusFilterItem: FilterItem[],
    sortBy: string
  ) => {
    props.setBrandFilters([...brandFilterItems]);
    props.setCategoryFilters([...categoryFilterItems]);
    props.setStatusFilters([...statusFilterItem]);
    props.setSortBy(sortBy);
  };

  const getNumberOfSelectedfilters = (filterItems: FilterItem[]) => filterItems.filter(item => item.isSelected).length;

  return (
    <div className={styles["filter-bar"]}>
      {props.screenSize && props.screenSize.viewType > ViewType.Tablet ? (
        <div className={`${styles["filter-bar__filter-container"]} ${"uk-container"}`}>
          <div className={`${styles["filter-bar__filter-list"]}  `}>
            <div className={`${styles["filter-label"]}`}>
              {props.intl.formatMessage({ id: "filter-bar-label" })}
              <span>
                <IconComponent icon={HandPointing} size={"20px"} />
              </span>
            </div>
            <div>
              <CheckboxDropdown
                title={props.intl.formatMessage({ id: "filter-bar-status-filter-title" })}
                onChange={onStatusFilterItemsChanged}
                items={[...props.statusFilterItems.map(item => ({ ...item }))]}
              />
            </div>
            <div>
              <SearchableCheckboxDropdown
                searchPlaceholder={props.intl.formatMessage(
                  { id: "filter-bar-search-placeholder" },
                  {
                    title: props.intl.formatMessage({ id: "filter-bar-category-filter-title" })
                  }
                )}
                deleteFilterLabel={props.intl.formatMessage(
                  { id: "filter-control-clear" },
                  {
                    title: props.intl.formatMessage({ id: "filter-bar-category-filter-title" })
                  }
                )}
                title={props.intl.formatMessage({ id: "filter-bar-category-filter-title" })}
                showFilterName={props.intl.formatMessage({ id: "filter-bar-categories" })}
                items={[...props.categoryFilterItems.map(item => ({ ...item }))]}
                onChange={onCategoryFilterItemsChanged}
              />
            </div>
            <div>
              <SearchableCheckboxDropdown
                searchPlaceholder={props.intl.formatMessage(
                  { id: "filter-bar-search-placeholder" },
                  {
                    title: props.intl.formatMessage({ id: "filter-bar-brand-filter-title" })
                  }
                )}
                deleteFilterLabel={props.intl.formatMessage(
                  { id: "filter-control-clear" },
                  { title: props.intl.formatMessage({ id: "filter-bar-brand-filter-title" }) }
                )}
                title={props.intl.formatMessage({ id: "filter-bar-brand-filter-title" })}
                showFilterName={props.intl.formatMessage({ id: "filter-bar-brand-filter-title" })}
                items={[...props.brandFilterItems.map(item => ({ ...item }))]}
                onChange={onBrandFilterItemsChanged}
              />
            </div>
            {getTotalNumberOfFilters() > 0 && (
              <div className={styles["filter-bar__clear-filter"]}>
                <span
                  role="link"
                  onClick={() => {
                    props.clearFilters();
                    props.filtersChanged();
                  }}
                  style={{ color: "red", cursor: "pointer" }}
                >
                  {props.intl.formatMessage({ id: "filter-bar-clear-filters" })}({getTotalNumberOfFilters()})
                </span>
              </div>
            )}
          </div>

          <div className={styles["filter-bar__sort"]}>
            <div className={styles["filter__sort-filter"]}>
              <span className={styles["amount-of-shops"]}>
                <IconComponent icon={StoreIcon} size={"20px"} />
              </span>
              {props.totalResults} {props.intl.formatMessage({ id: "tab-winkels" })}
            </div>
            <div className={styles["filter__sort-item"]}>
              {/* className={styles["filter__sort-change"]} */}
              {/* {orderBy} */}
              <SelectComponent
                options={["Relevatie", "Nieuwste", "Populair", "Winkels A - Z", "Winkels Z - A"]}
                onSelect={setSort}
              />
            </div>
          </div>
        </div>
      ) : (
        <StoresMobileFilterBarComponent
          brandFilterItems={props.brandFilterItems}
          categoryFilterItems={props.categoryFilterItems}
          sortBy={props.sortBy}
          sortByOptions={sortByOptions}
          statusFilterItems={props.statusFilterItems}
          onFiltersChanged={mobileFiltersChanged}
        />
      )}
    </div>
  );
};
const FilterBar = injectIntl(component);
export { FilterBar };
