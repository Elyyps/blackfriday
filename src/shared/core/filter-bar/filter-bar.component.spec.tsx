import * as React from "react";

import { storiesOf } from "@storybook/react";
import { withA11y } from "@storybook/addon-a11y";
import { FilterBarComponent } from "./filter-bar.component";
const notes = require("./filter-bar.md");

storiesOf("FilterBar", module)
  .addDecorator(withA11y)
  .add("Basic implementation", () => <FilterBarComponent />, {
    notes
  });
