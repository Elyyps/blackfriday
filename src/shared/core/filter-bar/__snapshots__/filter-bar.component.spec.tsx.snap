// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[FilterBar] should render component without crashing 1`] = `
<div
  className="filter-bar"
>
  <div
    className="filter-bar__filter-list"
  >
    <div
      className=" filter-label  uk-visible@m "
    >
      Filters
      <span>
        <IconComponent
          icon="hand-pointing.svg"
          size="20px"
        />
      </span>
    </div>
    <div
      className=" filter-label filter-label--mobile  uk-hidden@m"
      onClick={[Function]}
      role="button"
    >
      Filters
      <span>
        <IconComponent
          icon="filter.svg"
          size="16px"
        />
      </span>
    </div>
    <div
      className=" filter-content   "
    >
      <div
        className=" filter-content-label  uk-hidden@m"
      >
        <span
          className="button-control"
          onClick={[Function]}
          role="button"
        >
          <IconComponent
            color="#fff"
            icon="cross.svg"
            size="12px"
          />
        </span>
        <div>
          Filters
          <span
            className="filter-content-label-icon"
          >
            <IconComponent
              icon="hand-pointing.svg"
              size="16px"
            />
          </span>
        </div>
        <a
          className="filter-content-clear"
          onClick={[Function]}
          role="button"
        >
          Wis alle filters
        </a>
      </div>
    </div>
    <div
      className="uk-visible@m"
    >
      <DropdownComponent
        buttonColor="outline"
        buttonVariant="primary-brand"
        orientation="bottom-right"
        title="Status"
      >
        <ul
          className="filter-bar-ul"
        >
          <li
            className="filter-bar-li"
            key="0"
          >
            <CheckboxComponent
              isChecked={false}
              onClick={[Function]}
            >
              Actie nog onbekend
            </CheckboxComponent>
            (
            12
            )
          </li>
          <li
            className="filter-bar-li"
            key="1"
          >
            <CheckboxComponent
              isChecked={false}
              onClick={[Function]}
            >
              Actie start over 5 dagen!
            </CheckboxComponent>
            (
            12
            )
          </li>
          <li
            className="filter-bar-li"
            key="2"
          >
            <CheckboxComponent
              isChecked={false}
              onClick={[Function]}
            >
              Nu geldig
            </CheckboxComponent>
            (
            3
            )
          </li>
          <li
            className="filter-bar-li"
            key="3"
          >
            <CheckboxComponent
              isChecked={false}
              onClick={[Function]}
            >
              Bijna Verlopen!
            </CheckboxComponent>
            (
            3
            )
          </li>
          <li
            className="filter-bar-li"
            key="4"
          >
            <CheckboxComponent
              isChecked={false}
              onClick={[Function]}
            >
              Actie verlopen
            </CheckboxComponent>
            (
            3
            )
          </li>
        </ul>
      </DropdownComponent>
    </div>
    <div
      className="uk-visible@m"
    >
      <DropdownComponent
        buttonColor="outline"
        buttonVariant="secondary"
        orientation="bottom-left"
        title="Categories"
      >
        <SearchFilterControlComponent
          applyFilter={[MockFunction]}
          checkbox={
            Array [
              Object {
                "isChecked": false,
                "label": 12,
                "text": "Audio",
              },
              Object {
                "isChecked": false,
                "label": 3,
                "text": "Electronic",
              },
              Object {
                "isChecked": false,
                "label": 20,
                "text": "Sport",
              },
              Object {
                "isChecked": false,
                "label": 2,
                "text": "Mode",
              },
              Object {
                "isChecked": false,
                "label": 9,
                "text": "Games",
              },
            ]
          }
          getCheckedFilters={[Function]}
          getTotalShops={[Function]}
          numberOfFilters={0}
          onSelect={[MockFunction]}
        />
      </DropdownComponent>
    </div>
    <div
      className="uk-visible@m"
    >
      <DropdownComponent
        buttonColor="outline"
        buttonVariant="secondary"
        orientation="bottom-left"
        title="Brands"
      >
        <SearchFilterControlComponent
          applyFilter={[MockFunction]}
          checkbox={
            Array [
              Object {
                "isChecked": false,
                "label": 0,
                "text": "Appel ",
              },
              Object {
                "isChecked": false,
                "label": 255,
                "text": "Adidas",
              },
              Object {
                "isChecked": false,
                "label": 22,
                "text": "Converse",
              },
              Object {
                "isChecked": false,
                "label": 12,
                "text": "Guess",
              },
              Object {
                "isChecked": false,
                "label": 2,
                "text": "JBL",
              },
              Object {
                "isChecked": false,
                "label": 25,
                "text": "Only",
              },
              Object {
                "isChecked": false,
                "label": 0,
                "text": "Test1 ",
              },
              Object {
                "isChecked": false,
                "label": 255,
                "text": "Test2",
              },
              Object {
                "isChecked": false,
                "label": 22,
                "text": "Test3",
              },
              Object {
                "isChecked": false,
                "label": 12,
                "text": "Test4",
              },
              Object {
                "isChecked": false,
                "label": 2,
                "text": "Test5",
              },
              Object {
                "isChecked": false,
                "label": 25,
                "text": "Test6",
              },
              Object {
                "isChecked": false,
                "label": 12,
                "text": "Test7",
              },
              Object {
                "isChecked": false,
                "label": 2,
                "text": "Test8",
              },
              Object {
                "isChecked": false,
                "label": 25,
                "text": "Test9",
              },
            ]
          }
          getCheckedFilters={[Function]}
          getTotalShops={[Function]}
          numberOfFilters={0}
          onSelect={[MockFunction]}
        />
      </DropdownComponent>
    </div>
    <div
      className="uk-visible@m"
      style={
        Object {
          "marginLeft": "20px",
        }
      }
    >
      <span
        onClick={[Function]}
        role="link"
        style={
          Object {
            "color": "red",
            "cursor": "pointer",
          }
        }
      >
        Verwijder merk filters (
        0
        )
      </span>
    </div>
  </div>
  <div
    className="filter-bar__sort"
  >
    <div
      className="filter__sort-item"
    >
      <span>
        <IconComponent
          icon="store.svg"
          size="20px"
        />
      </span>
      0
       winkels
    </div>
    <div
      className="filter__sort-item"
    >
      Sorted by
      <span
        className="filter__sort-change"
        onClick={[Function]}
        role="button"
      >
        Relevant
        <span
          className=" filter__sort-change-icon  "
        >
          <IconComponent
            icon="arrow-long-down.svg"
            size="6px"
          />
        </span>
      </span>
      <ul
        className="filter__sort-ul"
        data-uk-dropdown="mode: click"
      >
        <li
          className="filter__sort-li"
          key="0"
          onClick={[Function]}
          role="link"
        >
          Newest
        </li>
        <li
          className="filter__sort-li"
          key="1"
          onClick={[Function]}
          role="link"
        >
          Relevant
          <IconComponent
            color="red"
            icon="check.svg"
            size="15px"
          />
        </li>
        <li
          className="filter__sort-li"
          key="2"
          onClick={[Function]}
          role="link"
        >
          Price
        </li>
      </ul>
    </div>
  </div>
</div>
`;
